addons/search-profile/userscript.js

```diff
@@ -5,14 +5,13 @@ export default async function ({ addon, console, msg }) {
     img = tab.appendChild(document.createElement("img")),
     span = tab.appendChild(document.createElement("span")),
     user = document.querySelector('[name="q"]').value.trim(),
-    valid = /^[\w-]{3,20}$/g.test(user);
+    valid = (/^[\w-]{3,20}$/g.test(user) && (await fetch(`https://scratch.mit.edu/users/${user}/`)).status == 200);
```

Comment:
Because of this `fetch`, it takes some time for the tab to appear. I think it would be better to show it immediately, then fetch this URL and disable the button element if it fails.